homeassistant/components/conversation/default_agent.py

```diff
@@ -187,12 +187,20 @@ async def async_recognize(
             _LOGGER.warning("No intents were loaded for language: %s", language)
             return None
 
+        intent_context = None
+
+        if user_input.device_id:
+            if device := dr.async_get(self.hass).async_get(user_input.device_id):
+                if (device is not None) and device.area_id:
+                    intent_context = {"area": device.area_id, "device_id": device.id}
```

Comment:
```suggestion
Can be collapsed in 1 statement:
```suggestion
        if user_input.device_id and device := dr.async_get(self.hass).async_get(user_input.device_id) and device.area_id:
            intent_context = {"area": device.area_id, "device_id": device.id}
```